contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,444 @@
+
+        uint256 oldDepositWeight = pcvDepositWeights[depositAddress];
+        int256 difference = newDepositWeight.toInt256() - oldDepositWeight.toInt256();
+        pcvDepositWeights[depositAddress] = newDepositWeight;
+
+        totalWeight = (totalWeight.toInt256() + difference).toUint256();
+
+        emit DepositWeightUpdate(depositAddress, oldDepositWeight, newDepositWeight);
+    }
+
+    /// @notice sets the weight of a pcv deposit to zero
+    /// @param depositAddress the address of the pcv deposit to set the weight of to zero
+    function setPCVDepositWeightZero(address depositAddress) external override onlyGuardianOrGovernor {
+        require(pcvDepositAddresses.contains(depositAddress), "Deposit does not exist.");
+
+        uint256 oldDepositWeight = pcvDepositWeights[depositAddress];
+        pcvDepositWeights[depositAddress] = 0;
+
+        totalWeight = (totalWeight.toInt256() - oldDepositWeight.toInt256()).toUint256();
+
+        emit DepositWeightUpdate(depositAddress, oldDepositWeight, 0);
+    }
+
+    /// @notice remove a PCV deposit from the set of deposits
+    /// @param pcvDeposit the address of the PCV deposit to remove
+    /// @param shouldWithdraw whether or not we want to withdraw from the pcv deposit before removing
+    function removePCVDeposit(address pcvDeposit, bool shouldWithdraw) external override onlyGovernorOrAdmin {
+        _removePCVDeposit(address(pcvDeposit), shouldWithdraw);
+    }
+
+    /// @notice adds a new PCV Deposit to the set of deposits
+    /// @param weight a relative (i.e. not normalized) weight of this PCV deposit
+    /// @dev the require check here is not in the internal method because the token var (as an immutable var) cannot be read in the constructor
+    function addPCVDeposit(address newPCVDeposit, uint256 weight) external override onlyGovernorOrAdmin {
+        require(IPCVDeposit(newPCVDeposit).balanceReportedIn() == token, "Deposit token must be the same as for this aggregator.");
+
+        _addPCVDeposit(newPCVDeposit, weight);
+    }
+
+    /// @notice replaces this contract with a new PCV Deposit Aggregator on the rewardsAssetManager
+    /// @param newAggregator the address of the new PCV Deposit Aggregator
+    function setNewAggregator(address newAggregator) external override onlyGovernor {
+        require(PCVDepositAggregator(newAggregator).token() == token, "New aggregator must be for the same token as the existing.");
+
+        // Add each pcvDeposit to the new aggregator
+        for (uint256 i=0; i < pcvDepositAddresses.length(); i++) {
+            address pcvDepositAddress = pcvDepositAddresses.at(i);
+            if (!(IPCVDepositAggregator(newAggregator).hasPCVDeposit(pcvDepositAddress))) {
+                uint256 pcvDepositWeight = pcvDepositWeights[pcvDepositAddress];
+                IPCVDepositAggregator(newAggregator).addPCVDeposit(pcvDepositAddress, pcvDepositWeight);
+            }
```

Comment:
I don't think this is needed. It means this contract has to be an admin of the new aggregator